apps/obs-web/src/views/AttributeSearchComp.vue

```diff
@@ -0,0 +1,239 @@
+            @filter="selectionFilterFn"
+            clearable
+        >
+        </q-select>
+        <div class="col" style="position: relative; top: 8px">
+            <q-spinner v-if="loading" color="primary" size="1.8em"></q-spinner>
+            <span v-if="selectedSearchResults.length > 0">{{selectedSearchResults.length}} results</span>
+        </div>
+      </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { createComponent, ref, onMounted, watch } from '@vue/composition-api';
+import Vue, { WatchOptions } from 'vue';
+import moment from 'moment';
+
+import { CouchDBInfo, CouchDBCredentials, couchService } from '@boatnet/bn-couch';
+import { Client, CouchDoc, ListOptions } from 'davenport';
+import Multiselect from 'vue-multiselect';
+
+Vue.component('multiselect', Multiselect);
+
+export default createComponent({
+  props: {
+    val: Array
+  },
+  setup(props, context) {
+    const store = context.root.$store;
+    const state = store.state;
+    const masterDB: Client<any> = couchService.masterDB;
+
+    const searchTypes: any = ref([]);
+    const selectedSearchType = ref('');
+    const searchOptions: any = ref([]);
+    const sourceSearchOptions: any = ref([]);
+    const selectedSearchOption = ref('');
+    const selectedSearchResults: any = ref([]);
+    const evaluator: any = ref('equals');
+    const loading: any = ref(false);
+
+    const getSearchTypes = async () => {
+      const results = await masterDB.view(
+        'obs_web',
+        'wcgop_trips_compound_fields',
+        {reduce: true, include_docs: false, group_level: 1} as any
+      );
+      searchTypes.value.push.apply(searchTypes.value, results.rows.map( (row: any) => row.key[0]));
+    };
+
+    const getSearchOptions = async () => {
+        loading.value = true;
+        sourceSearchOptions.value.length = 0;
+        const searchOptionResults = await masterDB.view(
+            'obs_web',
+            'wcgop_trips_compound_fields',
+            {reduce: true, include_docs: false, start_key: [selectedSearchType.value], end_key: [selectedSearchType.value, {}], group_level: 2} as any
+        );
+        sourceSearchOptions.value.push.apply(sourceSearchOptions.value, [...new Set(searchOptionResults.rows.map( (row: any) => {
+            if (row.key[1] === null || row.key[1] === undefined) {
+                return 'null';
+            } else if (['departureDate', 'returnDate'].includes(selectedSearchType.value)) {
+                return moment(row.key[1]).minute(0).hour(0).second(0).format();
+            } else {
+                return row.key[1];
+            }
+        }))]
+        );
+        loading.value = false;
+    };
+
+    const selectionFilterFn = (val: string, update: any, abort: any) => {
+        update(() => {
+            if (val !== '') {
+                searchOptions.value = sourceSearchOptions.value.filter( (option: any) => {
+                    return option.toString().toLowerCase().includes(val.toLowerCase());
```

Comment:
@melinashak-noaa seems like it would be useful to search for an observer by last name or logbook number by only the last few digits.  Do you think it's worth having a toggle for starts with / contains, or do we just pick one?

Reply:
oh I see, hmmmm we could probably can just keep it as is then with includes. If the debriefers think otherwise we can change it then.